app/src/main/java/com/hedvig/app/feature/home/ui/changeaddress/ChangeAddressActivity.kt

```diff
@@ -34,14 +52,118 @@ class ChangeAddressActivity : BaseActivity(R.layout.change_address_activity) {
+        Loading -> {
+            binding.spinner.loadingSpinner.show()
+            binding.contentScrollView.remove()
+        }
+        SelfChangeAddress -> setContent(
+            titleText = getString(R.string.moving_intro_title),
+            subtitleText = getString(R.string.moving_intro_description),
+            buttonText = getString(R.string.moving_intro_open_flow_button_text),
+            buttonIcon = null,
+            // TODO: Start Embark
+            onContinue = { }
+        )
+        ManualChangeAddress -> setContent(
+            titleText = getString(R.string.moving_intro_title),
+            subtitleText = getString(R.string.moving_intro_manual_handling_description),
+            buttonText = getString(R.string.moving_intro_manual_handling_button_text),
+            buttonIcon = R.drawable.ic_chat_white,
+            onContinue = { openChat() }
+        )
+        is ChangeAddressInProgress -> setUpcomingChangeContent(
+            titleText = getString(R.string.moving_intro_existing_move_title),
+            subtitleText = getString(R.string.moving_intro_existing_move_description),
+            buttonText = getString(R.string.moving_intro_manual_handling_button_text),
+            buttonIcon = R.drawable.ic_chat_white,
+            onContinue = { openChat() },
+            viewState.upcomingAgreementResult
+        )
+        is UpcomingAgreementError -> setContent(
+            titleText = getString(R.string.error_dialog_title),
+            subtitleText = when (viewState.error) {
+                NoContractsError -> "You do not have any contracts eligible for address change"
+                is GeneralError -> viewState.error.message ?: "Could not continue, please try again later"
+            },
+            buttonText = "Try again",
+            buttonIcon = null,
+            onContinue = { viewModel.reload() }
+        )
+        is SelfChangeError -> setContent(
+            titleText = getString(R.string.error_dialog_title),
+            subtitleText = viewState.error.message ?: "Could not continue, please try again later",
+            buttonText = "Try again",
+            buttonIcon = null,
+            onContinue = { viewModel.reload() }
+        )
+    }
+
+    private fun openChat() = startActivity(ChatActivity.newInstance(this, true))
+
+    private fun setContent(
+        titleText: String,
+        subtitleText: String,
+        buttonText: String?,
+        @DrawableRes buttonIcon: Int?,
+        onContinue: () -> Unit
+    ) = with(binding) {
+        spinner.loadingSpinner.remove()
+        contentScrollView.show()
+        image.show()
+        title.text = titleText
+        title.show()
+        subtitle.text = subtitleText
+        subtitle.show()
+        continueButton.text = buttonText
+        continueButton.setOnClickListener { onContinue() }
+        continueButton.icon = buttonIcon?.let { ContextCompat.getDrawable(this@ChangeAddressActivity, it) }
```

Comment:
```suggestion
        continueButton.icon = buttonIcon?.let { compatDrawable(it) }
```

There's a handy extension function for this purpose (needs to be imported, though 🙂